Rakefile

```
@@ -89,6 +89,37 @@ task :test do
   sh 'cargo test --workspace'
 end
 
+namespace :unicode do
+  ucd_dir = 'generated/ucd'
+
+  desc 'Rebuild Rust generated Rust sources from Unicode data'
+  task :build do
+    unless system('ucd-generate --version')
+      puts 'Please install ucd-generate > 0.3.0 first'
+      exit(1)
+    end
+
+    unless system('git diff --exit-code')
```

I'm wondering if something like this to be a bit more specific to the changes, however I see the later cargo clippy, and couldn't find a way to make that run specifically in the directory, so maybe a full dirty check is just better. Just leaving the suggestion in case other things come to mind.

```suggestion
    unless system('git diff --exit-code -- "generated/**/*"')
```

---

> couldn't find a way to make that run specifically in the directory

So do I. Seem like the only way is to have a separate package (`cargo clippy -p ...`).

The motivation for this check was to prevent the later `cargo clippy --fix` from modifying unstaged work unexpectedly. In contrast, I'm not worried too much about modifying `generated/**/*`, since they can be easily re-generated.